### ISSUE TITLE
Problem with app extension. Wrong order id

### ISSUE THREAD

**Issue by JulienSansot** [Shopify/shopify_app#1731]
Hi,
I've been trying to use an App Extension but I keep having a strange problem.
The link goes to the wrong order ID.

I have an App extension setup like so:
<img width="652" alt="image" src="https://github.com/Shopify/shopify_app/assets/482806/82ca68f0-a994-4790-b466-936eefaeed65">

However, the "id" parameters I get after the first one are always one step behind
For example:
if I'm looking at order 111 and I click on the app extension link, I receive `id=111` in my app, all good.
Then, if I go to order 222, and click on the app extension link, I receive `id=111` in my app, (instead of 222)
Then, if I go to order 333, and click on the app extension link, I receive `id=222` in my app, (instead of 333)
Then, if I go to order 444, and click on the app extension link, I receive `id=333` in my app, (instead of 444)
etc...

The `id` parameter I receive is always the previous one.

Another strange thing is that the issue only happens on the deployed version of the app, but not when I run it in local.

- `shopify_api` version: 12.5.0
- `shopify_app` version: 21.6.0
- Ruby version: 3.2.1
- Operating system: debian slim buster

**Comment by ptSquare** [comment]
I'm running into the same issue but with the Customer Extension. It keeps going to the previously selected customer. It does go to the correct customer when you open a new tab though, so the link itself seems to be good. 

It's working fine on our test versions locally and deployed, but this problem is only happening with the live version of the app.

**Comment by ptSquare** [comment]
The problem for us turned out be something to do with the hosting. Deleted and relaunched on new machines and it's working fine now.

**Comment by esschul** [comment]
We are seeing the exact same issue, did you figure this one out?

**Comment by JulienSansot** [comment]
> We are seeing the exact same issue, did you figure this one out?

No, but I know that Shopify is working on a fix. They said it should be fixed this week. 🤞

**Comment by esschul** [comment]
Great, where did you get information about them working on it?

**Comment by nelsonwittwer** [comment]
👋 Can you provide more information on how you concluded the issue was with the `shopify_app` gem? It seems like this is an issue somewhere else in the platform but happy to help out if that doesn't appear to the case!

**Comment by JulienSansot** [comment]
> 👋 Can you provide more information on how you concluded the issue was with the `shopify_app` gem? It seems like this is an issue somewhere else in the platform but happy to help out if that doesn't appear to the case!

Hi @nelsonwittwer,
to be honest, I'm really not sure it's something to do with the `shopify_app` gem. But I concluded that because:
1. I didn't find anyone else complaining about the issue so I thought it must be something to do with the way my app is built rather than with all app extensions.
2. I couldn't find any issue with my code. I made is super simple, just a basic rails route going to an empty page but could see that the log was still showing the incorrect param.
3. The link clicked by the user is correct so I assumed it's not a Shopify's platform problem.
